lib/core-classes/AssetCategoryService.ts

```diff
@@ -235,11 +239,17 @@ export class AssetCategoryService {
   }
 
   formatMetadataForGet(assetMetadata: FormattedMetadata[]) {
-    const formattedMetadata: JSONObject = {};
-    for (const metadata of assetMetadata) {
-      formattedMetadata[metadata.key] = this.getValue(metadata.value);
+    if (assetMetadata) {
+      if (!Array.isArray(assetMetadata)) {
+        return assetMetadata;
+      }
+      const formattedMetadata: JSONObject = {};
+      for (const metadata of assetMetadata) {
+        formattedMetadata[metadata.key] = this.getValue(metadata.value);
+      }
+      return formattedMetadata;
     }
-    return formattedMetadata;
+    return {};
   }
```

Comment:
Use fast return are more simplier to understand and most efficient
Source: https://szymonkrajewski.pl/why-should-you-return-early/

```ts
formatMetadataForGet(assetMetadata: FormattedMetadata[]) {
  if (!assetMetadata) {
    return {};
  }

  if (!Array.isArray(assetMetadata)) {
    return assetMetadata;
  }

  const formattedMetadata: JSONObject = {};
  for (const metadata of assetMetadata) {
    formattedMetadata[metadata.key] = this.getValue(metadata.value);
  }
  
  return formattedMetadata;
}
```

Reply:
I know. I got a very bad habit.

Reply:
done